perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/MembersManagerBlImpl.java

```diff
@@ -2398,7 +2398,7 @@ protected Pair<Boolean, Date> manageMembershipExpiration(PerunSession sess, Memb
 						fieldAmount = Utils.prepareGracePeriodDate(m);
 						LocalDate gracePeriodDate = localDate.minus(fieldAmount.getLeft(), fieldAmount.getRight());
 						// Check if we are in grace period
-						if (gracePeriodDate.isBefore(LocalDate.now())) {
+						if (gracePeriodDate.isEqual(LocalDate.now()) || gracePeriodDate.isBefore(LocalDate.now())) {
```

Comment:
is there a reason to not add it also to line 1984? and I think the identical problem is also in group membership grace periods in GroupsManager methods `isMemberInGracePeriod` and `extendForStaticDate`

Reply:
You're right; I did a sloppy job checking the use cases.